src/resolvers/deck/createDeck.ts

```diff
@@ -50,38 +50,44 @@ export const importDeck = mutationWithClientMutationId({
     { user, publishedDeckLoader, modelLoader, fieldsByModelLoader }: Context
   ) => {
     const helperMap: {
-      updateModelIndex: { oldId: string; newId: string }[]
-      updateFieldIndex: { oldId: string; newId: string }[]
+      updateModelIndex: Record<string, string>
+      updateFieldIndex: Record<string, string>
     } = {
-      updateModelIndex: [],
-      updateFieldIndex: [],
+      updateModelIndex: {},
+      updateFieldIndex: {},
     }
     const { id: deckId } = fromGlobalId(id)
     const publishedDeck = await publishedDeckLoader.load(deckId)
+
+    if (!publishedDeck) {
+      return
```

Comment:
this should actually return a null deck, else it will raise a runtime error

```suggestion
      return { deck: null }
```